python/cudf/cudf/io/parquet.py

```diff
@@ -203,6 +205,51 @@ def read_parquet(
             )
         filepaths_or_buffers.append(tmp_source)
 
+    if filters is not None:
+        # Convert filters to ds.Expression
+        filters = ioutils.filters_to_expression(filters)
+
+        # Initialize ds.FilesystemDataset
+        dataset = ds.dataset(filepaths_or_buffers, format="parquet")
+
+        # Load IDs of filtered row groups for each file in dataset
+        filtered_rg_ids = {}
+        for fragment in dataset.get_fragments(filter=filters):
+            for rg_fragment in fragment.get_row_group_fragments(filters):
+                for rg_id in rg_fragment.row_groups:
+                    path = rg_fragment.path
+                    if path not in filtered_rg_ids:
+                        filtered_rg_ids[path] = [rg_id]
+                    else:
+                        filtered_rg_ids[path].append(rg_id)
```

Comment:
Instead of having to always check this you can create `filtered_rg_ids` as a `defaultdict(list)` which will allow you to always call `append`